// scathach-api/build/src/consts.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const BASEURL = 'https://scathach.redsplit.org/v5';
const BASEARMORED = 'https://scathach.redsplit.org/api/armored';
exports.default = {
    endpoint: {
        fgo: `${BASEURL}/fgo/`,
        genshin: `${BASEURL}/genshin_impact/`,
        azur: `${BASEURL}/azur_lane/`,
        waifu: `${BASEURL}/waifu/`,
        arknights: `${BASEURL}/arknights/`,
        fire_emblem: `${BASEURL}/fire_emblem/`,
        gfl: `${BASEURL}/gfl/`,
        hololive: `${BASEURL}/hololive/`,
        kancolle: `${BASEURL}/kancolle/`,
        sex: `${BASEURL}/sex/`,
        gelbooru: `${BASEARMORED}/gelbooru`,
        r34: `${BASEARMORED}/r34`,
        safe: `${BASEARMORED}/safebooru`,
        reaction: ['angry', 'bonk', 'cry', 'happy', 'hug', 'kiss', 'lol', 'nom', 'pat', 'pout', 'smug', 'uwu', 'wink'],
        animesex: [
            'ass',
            'bdsm',
            'cum',
            'creampie',
            'manga',
            'femdom',
            'hentai',
            'incest',
            'masturbation',
            'public',
            'ero',
            'orgy',
            'elves',
            'yuri',
            'pantsu',
            'glasses',
            'cuckold',
            'blowjob',
            'boobjob',
            'foot',
            'thighs',
            'vagina',
            'ahegao',
            'uniform',
            'gangbang',
            'tentacles',
            'gif',
            'neko',
            'nsfwMobileWallpaper',
            'zettaiRyouiki'
        ]
    },
};
